src/Interfaces/hamiltonians.jl

```diff
@@ -419,3 +449,47 @@ See also [`LOStructure`](@ref Main.Hamiltonians.LOStructure).
 has_adjoint(op) = has_adjoint(LOStructure(op))
 has_adjoint(::AdjointUnknown) = false
 has_adjoint(::LOStructure) = true
+
+"""
+    OperatorColumn(op::AbstractOperator{T}, address::A) <: AbstractVector{Pair{A,T}}
+
+Default column, using `offdiagonals(op, address)` and `diagonal_element(op, address)`.
+"""
+struct OperatorColumn{A,T,O<:Union{AbstractOperator{T},AbstractMatrix{T}},OD} <: AbstractVector{Pair{A,T}}
```

Comment:
```suggestion
"""
    OffdiagonalsOperatorColumn(op::AbstractOperator{T}, address::A)

Default column, using [`offdiagonals(op, address)`](@ref) and [`diagonal_element(op, address)`](@ref).

See also [`operator_column`](@ref).
"""
struct OffdiagonalsOperatorColumn{A,T,O<:Union{AbstractOperator{T},AbstractMatrix{T}},OD} 
```
I think it is valuable to change the name as suggested in order to not imply that this is the only implementation of an operator column. In fact, I see this as a temporary type, as it is not efficient to separately call `offdiagonals` and `diagonal_element`.

I'm aware that it is a change of mind to not subtype the column to `AbstractVector`. I believe that it is better this way for encouraging the use of the (possibly more efficient) function of the documented interface.